contrib/fujifilm_dynamic_range.lua

```diff
@@ -0,0 +1,115 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+]]
+
+--[[About this Plugin
+Support for adjusting darktable exposure by Fujifilm raw exposure
+bias. This corrects for a DR100/DR200/DR400 "dynamic range" setting.
+
+Dependencies:
+- exiftool (https://www.sno.phy.queensu.ca/~phil/exiftool/)
+
+Based upon fujifilm_ratings by Ben Mendis
+
+The relevant tag is RawExposureBias (0x9650). This appears to
+represent the shift in EV for the chosen DR setting (whether manual or
+automatic). Note that even at 100DR ("standard") there is an EV shift:
+
+100 DR -> -0.72 EV
+200 DR -> -1.72 EV
+400 DR -> -2.72 EV
+
+The ideal would be to use exiv2 to read this tag, as this is the same
+code which darktable import uses. Unfortunately, exiv2 as of v0.27.3
+can't read this tag. As it is encoded as a 4-byte ratio of two signed
+shorts -- a novel data type -- it will require some attention to fix
+this.
+
+There is an exiv2-readable DevelopmentDynamicRange tag which maps to
+RawExposureBias as above.  DevelopmentDynamicRange is only present
+when tag DynamicRangeSetting (0x1402) is Manual/Raw (0x0001). When it
+is Auto (0x0000), the equivalent data is tag AutoDynamicRange
+(0x140b). But exiv2 currently can't read that tag either.
+
+Hence for now this code uses exiftool to read RawExposureBias, as a
+more general solution. As exiftool is approx. 10x slower than exiv2
+(Perl vs. C++), this may slow large imports.
+
+These tags have been checked on a Fujifilm X100S and X100V. Other
+cameras may behave in other ways.
+
+--]]
+
+local dt = require "darktable"
+local du = require "lib/dtutils"
+local df = require "lib/dtutils.file"
+local gettext = dt.gettext
+
+du.check_min_api_version("4.0.0", "fujifilm_dynamic_range")
+
+gettext.bindtextdomain("fujifilm_dynamic_range", dt.configuration.config_dir.."/lua/locale/")
+
+local function _(msgid)
+	return gettext.dgettext("fujifilm_dynamic_range", msgid)
+end
+
+local function detect_dynamic_range(event, image)
+	if image.exif_maker ~= "FUJIFILM" then
+		dt.print_log(_("[fujifilm_dynamic_range] ignoring non-Fujifilm image"))
+		return
+	end
+	-- it would be nice to check image.is_raw but this appears to not yet be set
+	if not string.match(image.filename, "%.RAF$") then
+		dt.print_log(_("[fujifilm_dynamic_range] ignoring non-raw image"))
+		return
+	end
+	if not df.check_if_bin_exists("exiftool") then
+		dt.print_error(_("[fujifilm_dynamic_range] exiftool not found"))
+		return
+	end
+	local RAF_filename = df.sanitize_filename(tostring(image))
+	-- without -n flag, exiftool will round to the nearest tenth
+	local command = "exiftool -RawExposureBias -n -t " .. RAF_filename
```

Comment:
df.check_if_bin_exists() returns the full path to the executable.  Using the returned value would let the script run on windows and macos too.

Reply:
Done! (Though I can only test on Linux...)